ui/src/app/views/notebook-redirect/component.ts

```diff
@@ -146,12 +228,14 @@ export class NotebookRedirectComponent implements OnInit, OnDestroy {
     return this.localizeNotebooks([]).flatMap((localDir) => {
       // Use the Jupyter Server API directly to create a new notebook. This
       // API handles notebook name collisions and matches the behavior of
-      // clicking "new notebook" in the Jupyter UI.
+      // clicking 'new notebook' in the Jupyter UI.
       const workspaceDir = localDir.replace(/^workspaces\//, '');
-      return this.jupyterService.postContents(
+      return this.jupyterService.putContents(
         this.cluster.clusterNamespace, this.cluster.clusterName,
-        workspaceDir, {
-          'type': 'notebook'
+        workspaceDir, this.notebookName + '.ipynb', {
+          'type': 'file',
```

Comment:
Wanted to confirm that what you were trying included passing format "json" instead of "text". 

```
{
  'type': 'notebook',
  'format': 'json',
  'content': { ... } 
}
```

Reply:
Yeah, just tried passing notebook as type, with format JSON, and got: 
`{"message": "Unexpected error while saving file: workspaces/test/Test NB.ipynb 'unicode' object has no attribute 'nbformat'", "reason": null}` again. (The name clash thing that I mentioned.)

Reply:
And type text does not accept json as a legal field.

Reply:
That error makes it sound like you may have passed a serialized JSON string. I don't feel strongly, but would be slightly more natural with type "json" passing json.

Reply:
Content currently just accepts string arguments, not anything, so it doesn't really allow passing in as an object. I think type is a specific enum on the JupyterApi side.

On the error, we actually did a bunch of googling trying to figure it out, and it sounded like it was an explicit name collision that caused it, and the error message was just really painfully unclear...